examples/acoustic/acoustic_example.py

```diff
@@ -24,33 +24,34 @@ def source(t, f0):
     return (1-2.*r**2)*np.exp(-r**2)
 
 
-def run(dimensions=(150, 150, 50), spacing=(20.0, 20.0, 20.0), tn=250.0,
-        time_order=2, space_order=2, nbpml=10, cse=True, auto_tuning=False,
+def run(dimensions=(50, 50, 50), spacing=(20.0, 20.0, 20.0), tn=1000.0,
+        time_order=2, space_order=2, nbpml=40, cse=True, auto_tuning=False,
         compiler=None, cache_blocking=None, full_run=False):
-    model = IGrid()
-    model0 = IGrid()
-    model1 = IGrid()
-    model.shape = dimensions
-    model0.shape = dimensions
-    model1.shape = dimensions
+
     origin = (0., 0., 0.)
 
     # True velocity
-    true_vp = np.ones(dimensions) + 2.0
-    true_vp[:, :, int(dimensions[0] / 2):int(dimensions[0])] = 4.5
+    true_vp = np.ones(dimensions) + .5
+    if len(dimensions) == 2:
+        true_vp[:, int(dimensions[0] / 2):dimensions[0]] = 2.5
+    else:
+        true_vp[:, :, int(dimensions[0] / 2):dimensions[0]] = 2.5
 
     # Smooth velocity
     initial_vp = smooth10(true_vp, dimensions)
 
     dm = 1. / (true_vp * true_vp) - 1. / (initial_vp * initial_vp)
 
-    model.create_model(origin, spacing, true_vp)
+    model = IGrid(origin, spacing, true_vp)
 
     # Define seismic data.
     data = IShot()
 
     f0 = .010
-    dt = model.get_critical_dt()
+    if time_order == 4:
+        dt = 1.73 * model.get_critical_dt()
+    else:
+        dt = model.get_critical_dt()
```

Comment:
And as I said, this is usage dependent. Can run 2nd order forward then 4th order gradient. This can't be generalized.